packages/target-ethers-v5/src/codegen/index.ts

```diff
@@ -0,0 +1,268 @@
+import { values } from 'lodash'
+import {
+  AbiParameter,
+  BytecodeWithLinkReferences,
+  Contract,
+  EventArgDeclaration,
+  EventDeclaration,
+  FunctionDeclaration,
+} from 'typechain'
+import { generateInputType, generateInputTypes } from './types'
+import { codegenFunctions } from './functions'
+
+export function codegenContractTypings(contract: Contract) {
+  const contractImports: string[] = ['Contract', 'ContractTransaction']
+  const allFunctions = values(contract.functions)
+    .map(codegenFunctions.bind(null, { returnResultObject: true }))
+    .join('')
+
+  if (allFunctions.match(/\W Overrides(\W|$)/)) contractImports.push('Overrides')
+  if (allFunctions.match(/\WPayableOverrides(\W|$)/)) contractImports.push('PayableOverrides')
+  if (allFunctions.match(/\WCallOverrides(\W|$)/)) contractImports.push('CallOverrides')
```

Comment:
why is the first regex different than the 2nd & 3rd one? (space between `\W` and the name)

Reply:
maybe is it worth extracting a helper fn to help readability and avoid typos in the regexes? like
```typescript
function pushImportIfUsed(importName: string, generatedCode: string, importArray: string[]): void {
  if (new RegExp(`\\W${importName}(\\W|$)`).test(generatedCode)) importArray.push(importName);
}
```

Reply:
I was afraid that `/\WOverrides(\W|$)/` would match `PayableOverrides` and `CallOverrides` too when it shouldn't and should only match `Overrides`. Is there a better way to do the same? I think would face the same issue having this as a method as the `pushImportIfUsed` example

Reply:
I've noticed that problem when I wrote those regexes in the `ethers-v4` target :) `\W` means a non-word character, and it's there specifically so `xyz` doesn't match `abcxyz`, so it's safe to drop the extra space.

Reply:
Example: https://regex101.com/r/r6Cjn4/1